From d77da7543a1373fd0d78684461e945647bb63e41 Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@freescale.com>
Date: Tue, 23 Jul 2013 15:17:14 +0300
Subject: [PATCH 098/383] dpaa_eth: Fix freeing of bpool buffers

Make sure buffers get freed correctly even if they are part of
compound pages.

Signed-off-by: Ioana Radulescu <ruxandra.radulescu@freescale.com>
Change-Id: I36aaa1d13b526fa34c7c6ac7d3a1d787091b50f2
Reviewed-on: http://git.am.freescale.net:8181/3481
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Hamciuc Bogdan-BHAMCIU1 <bogdan.hamciuc@freescale.com>
Reviewed-by: Bucur Madalin-Cristian-B32716 <madalin.bucur@freescale.com>
Reviewed-by: Sovaiala Cristian-Constantin-B39531 <Cristian.Sovaiala@freescale.com>
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
[Original patch taken from QorIQ-SDK-V1.5-SOURCE-20131219-yocto.iso]
Signed-off-by: Bin Jiang <bin.jiang@windriver.com>
---
 drivers/net/ethernet/freescale/dpa/dpaa_eth.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth.h b/drivers/net/ethernet/freescale/dpa/dpaa_eth.h
index 31f7052..d1304c1 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth.h
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth.h
@@ -713,7 +713,7 @@ static inline void _dpa_bp_free_buf(void *addr)
 #else
 static inline void _dpa_bp_free_buf(void *addr)
 {
-	free_page((unsigned long)addr);
+	put_page(virt_to_head_page(addr));
 }
 #endif
 
-- 
1.7.5.4

